hostsfile/hostsfile_test.go

```diff
@@ -74,30 +105,78 @@ func TestParseLine(t *testing.T) {
 	}
 
 	// Not Commented stuff
-	hosts = parseLine("255.255.255.255 broadcasthost test.domain.com	domain.com")
-	if !hosts.Contains(newHostname("broadcasthost", net.ParseIP("255.255.255.255"), false)) ||
-		!hosts.Contains(newHostname("test.domain.com", net.ParseIP("255.255.255.255"), false)) ||
-		!hosts.Contains(newHostname("domain.com", net.ParseIP("255.255.255.255"), false)) ||
+	hosts = parseLine("192.168.0.1 broadcasthost test.domain.com	domain.com")
+	if !hosts.Contains(newHostname("broadcasthost", net.ParseIP("192.168.0.1"), false, false)) ||
+		!hosts.Contains(newHostname("test.domain.com", net.ParseIP("192.168.0.1"), false, false)) ||
+		!hosts.Contains(newHostname("domain.com", net.ParseIP("192.168.0.1"), false, false)) ||
 		len(hosts) != 3 {
 		t.Error("Expected to find broadcasthost, domain.com, and test.domain.com")
 	}
 
+	// Wildcard stuff
+	hosts = parseLine("192.168.0.1 *.domain.com mail.domain.com serenity")
+	if !hosts.Contains(newHostname(".domain.com", net.ParseIP("192.168.0.1"), false, true)) ||
+		!hosts.Contains(newHostname("mail.domain.com", net.ParseIP("192.168.0.1"), false, false)) ||
+		!hosts.Contains(newHostname("serenity", net.ParseIP("192.168.0.1"), false, false)) ||
+		len(hosts) != 3 {
+		t.Error("Expected to find *.domain.com, mail.domain.com and serenity.")
+	}
+
+	var ip net.IP;
+
+	ip = hosts.FindHost("api.domain.com");
+	if !net.ParseIP("192.168.0.1").Equal(ip) {
+		t.Error("Can't match wildcard host api.domain.com");
+	}
+
+	ip = hosts.FindHost("google.com")
+	if ip != nil {
+		t.Error("We shouldn't resolve google.com");
+	}
+
+	hosts = *newHostlistString(`192.168.0.1 *.domain.com mail.domain.com serenity
+				192.168.0.2	api.domain.com`);
+
+	if (!net.ParseIP("192.168.0.2").Equal(hosts.FindHost("api.domain.com"))) {
+		t.Error("Failed matching api.domain.com explicitly");
+	}
+	if (!net.ParseIP("192.168.0.1").Equal(hosts.FindHost("mail.domain.com"))) {
+		t.Error("Failed matching api.domain.com explicitly");
+	}
+	if (!net.ParseIP("192.168.0.1").Equal(hosts.FindHost("wildcard.domain.com"))) {
+		t.Error("Failed matching wildcard.domain.com explicitly");
+	}
+	if (net.ParseIP("192.168.0.1").Equal(hosts.FindHost("sub.wildcard.domain.com"))) {
+		t.Error("Failed not matching sub.wildcard.domain.com explicitly");
+	}
+
+	// IPv6 (not link-local)
+	hosts = parseLine("2a02:7a8:1:250::80:1		rtvslo.si img.rtvslo.si")
+	if !hosts.Contains(newHostname("img.rtvslo.si", net.ParseIP("2a02:7a8:1:250::80:1"), true, false)) ||
+		len(hosts) != 2 {
+		t.Error("Expected to find rtvslo.si ipv6, two hosts")
+	}
+
+	/* the following all fails since the addressses are link-local */
+
```

Comment:
Yeah....the loopback addresses are not parsed because they are matched by the `net.IsGlobalUnicast()` func. https://golang.org/src/net/ip.go?s=4290:4325#L146
We should probably allow them as it is technically legal to return them in a DNS answer. But this will be another PR.